File: web/src/utils/quoteFilters.ts
/**
 * Drop filler quotes from PR-facing UI (aligned with backend `is_trivial_quote`).
 */
export function isTrivialQuote(text: string, minLen = 24): boolean {
  const t = text.trim();
  if (!t) return true;
  if (t.length < minLen) return true;
  const lower = t.toLowerCase().replace(/\s+/g, ' ');
  const oneLine = lower.replace(/[.!?…]+$/u, '');
  if (
    /^(?:спасибо|благодарю|окей|ок|да+|нет+|супер|класс|понятно|хорошо|нормально|thanks|thank you|ok|cool|great|nice)(?:\s*[!.]*)?$/iu.test(
      oneLine,
    )
  ) {
    return true;
  }
  if (t.split(/\s+/u).length <= 2 && t.length < 40) return true;
  return false;
}

File: app/utils/quote_filters.py
"""Filter low-information quotes from PR-facing outputs."""

from __future__ import annotations

import re

# Very short pleasantries / fillers — not evidence for PR narrative.
_TRIVIAL_QUOTE_PATTERNS = re.compile(
    r"^(?:спасибо|благодарю|окей|ок|да+|нет+|супер|класс|понятно|хорошо|нормально|"
    r"thanks|thank you|ok\.?|cool|great|nice)\s*[!.]*$",
    re.IGNORECASE,
)


def is_trivial_quote(text: str, *, min_len: int = 24) -> bool:
    t = (text or "").strip()
    if len(t) < min_len:
        return True
    if _TRIVIAL_QUOTE_PATTERNS.match(t):
        return True
    if len(t.split()) <= 2 and len(t) < 40:
        return True
    return False
